### PR TITLE
Enable make check and sanity check exec for OpenMPI

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _OpenMPI-2.1.2-GCC-6.4.0-2.28.eb_ 
 * **SUCCESS** _OpenMPI-3.1.0-GCC-7.3.0-2.30.eb_ 
 * **SUCCESS** _OpenMPI-3.1.1-GCC-7.3.0-2.30.eb_ 
 * **SUCCESS** _OpenMPI-3.1.2-GCC-8.2.0-2.31.1.eb_ 
 * **SUCCESS** _OpenMPI-3.1.3-GCC-8.2.0-2.31.1.eb_ 
 * **SUCCESS** _OpenMPI-4.0.3-GCC-9.3.0.eb_ 
 * **SUCCESS** _OpenMPI-4.0.5-GCC-10.2.0.eb_ 
 * **SUCCESS** _OpenMPI-4.0.5-GCC-9.3.0.eb_ 
 * **SUCCESS** _OpenMPI-4.1.0-GCC-10.2.0.eb_ 
 * **SUCCESS** _OpenMPI-4.1.1-GCC-10.3.0.eb_ 
 * **SUCCESS** _GCC-8.3.0-2.32.eb_ 
 * **SUCCESS** _hwloc-2.0.2-GCCcore-8.2.0.eb_ 
 * **SUCCESS** _hwloc-2.0.3-GCCcore-8.3.0.eb_ 
 * **SUCCESS** _OpenMPI-3.1.4-GCC-8.3.0-2.32.eb_ 
 * **FAIL (build issue)** _OpenMPI-4.0.0-GCC-8.2.0-2.31.1.eb_ (partial log available at https://gist.github.com/69d8ea020c6cf5ec1bd07c40270bacff)
 * **FAIL (build issue)** _OpenMPI-4.0.1-GCC-8.3.0-2.32.eb_ (partial log available at https://gist.github.com/b8b6e37d65ac12d86fde58e57244233c)
 * **SUCCESS** _binutils-2.29.eb_ 
 * **SUCCESS** _GCCcore-9.2.0.eb_ 
 * **SUCCESS** _zlib-1.2.11-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _GCCcore-7.2.0.eb_ 
 * **SUCCESS** _zlib-1.2.11-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _help2man-1.47.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _help2man-1.47.10-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _M4-1.4.18-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _M4-1.4.18-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Bison-3.0.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Bison-3.3.2-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Bison-3.0.5-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _flex-2.6.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _flex-2.6.4-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _binutils-2.29-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _binutils-2.32-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _XZ-5.2.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _XZ-5.2.4-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Autoconf-2.69-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Autoconf-2.69-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _GCC-7.2.0-2.29.eb_ 
 * **SUCCESS** _GCC-9.2.0-2.32.eb_ 
 * **SUCCESS** _libxml2-2.9.8-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _libxml2-2.9.10-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Automake-1.15.1-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Automake-1.16.1-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _libtool-2.4.6-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _libtool-2.4.6-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Autotools-20170619-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Autotools-20180311-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _xorg-macros-1.19.2-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _xorg-macros-1.19.2-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _numactl-2.0.11-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _numactl-2.0.13-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _libpciaccess-0.14-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _libpciaccess-0.16-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _hwloc-1.11.8-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _hwloc-2.1.0-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _OpenMPI-3.0.0-GCC-7.2.0-2.29.eb_ 
 * **FAIL (build issue)** _OpenMPI-4.0.2-GCC-9.2.0-2.32.eb_ (partial log available at https://gist.github.com/cacce2894211f4b5cba908e101990f37)

Build succeeded for 53 out of 56 (15 easyconfigs in total)
taurusi6285.taurus.hrsk.tu-dresden.de - Linux RHEL 7.9, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/685db43818c3c597239e3ab9770f1c2e for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _OpenMPI-2.1.2-GCC-6.4.0-2.28.eb_ 
 * **SUCCESS** _OpenMPI-3.1.0-GCC-7.3.0-2.30.eb_ 
 * **SUCCESS** _OpenMPI-3.1.1-GCC-7.3.0-2.30.eb_ 
 * **SUCCESS** _OpenMPI-3.1.2-GCC-8.2.0-2.31.1.eb_ 
 * **SUCCESS** _OpenMPI-3.1.3-GCC-8.2.0-2.31.1.eb_ 
 * **SUCCESS** _OpenMPI-4.0.3-GCC-9.3.0.eb_ 
 * **SUCCESS** _OpenMPI-4.0.5-GCC-10.2.0.eb_ 
 * **SUCCESS** _OpenMPI-4.0.5-GCC-9.3.0.eb_ 
 * **SUCCESS** _OpenMPI-4.1.0-GCC-10.2.0.eb_ 
 * **SUCCESS** _OpenMPI-4.1.1-GCC-10.3.0.eb_ 
 * **SUCCESS** _GCC-8.3.0-2.32.eb_ 
 * **SUCCESS** _hwloc-2.0.2-GCCcore-8.2.0.eb_ 
 * **SUCCESS** _hwloc-2.0.3-GCCcore-8.3.0.eb_ 
 * **SUCCESS** _OpenMPI-3.1.4-GCC-8.3.0-2.32.eb_ 
 * **FAIL (build issue)** _OpenMPI-4.0.0-GCC-8.2.0-2.31.1.eb_ (partial log available at https://gist.github.com/eeb59036cd016497c075bffe16e110f0)
 * **FAIL (build issue)** _OpenMPI-4.0.1-GCC-8.3.0-2.32.eb_ (partial log available at https://gist.github.com/884ce4dda437727956f4c40a159140b2)
 * **SUCCESS** _binutils-2.29.eb_ 
 * **SUCCESS** _GCCcore-9.2.0.eb_ 
 * **SUCCESS** _zlib-1.2.11-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _GCCcore-7.2.0.eb_ 
 * **SUCCESS** _zlib-1.2.11-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _help2man-1.47.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _help2man-1.47.10-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _M4-1.4.18-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _M4-1.4.18-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Bison-3.0.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Bison-3.3.2-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Bison-3.0.5-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _flex-2.6.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _flex-2.6.4-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _binutils-2.29-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _binutils-2.32-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _XZ-5.2.4-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _XZ-5.2.4-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Autoconf-2.69-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Autoconf-2.69-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _GCC-7.2.0-2.29.eb_ 
 * **SUCCESS** _GCC-9.2.0-2.32.eb_ 
 * **SUCCESS** _libxml2-2.9.8-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _libxml2-2.9.10-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Automake-1.15.1-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Automake-1.16.1-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _libtool-2.4.6-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _libtool-2.4.6-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _Autotools-20170619-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _Autotools-20180311-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _xorg-macros-1.19.2-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _xorg-macros-1.19.2-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _numactl-2.0.11-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _numactl-2.0.13-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _libpciaccess-0.14-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _libpciaccess-0.16-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _hwloc-1.11.8-GCCcore-7.2.0.eb_ 
 * **SUCCESS** _hwloc-2.1.0-GCCcore-9.2.0.eb_ 
 * **SUCCESS** _OpenMPI-3.0.0-GCC-7.2.0-2.29.eb_ 
 * **FAIL (build issue)** _OpenMPI-4.0.2-GCC-9.2.0-2.32.eb_ (partial log available at https://gist.github.com/3a2b036fcfde287ee25602a0da8eb65c)

Build succeeded for 53 out of 56 (15 easyconfigs in total)
taurusi6283.taurus.hrsk.tu-dresden.de - Linux RHEL 7.9, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/97628b17ed4fda2ea9df0439e62ad069 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _OpenMPI-4.0.2-GCC-9.2.0-2.32.eb_ (partial log available at https://gist.github.com/ca4ef2a2236da9ba04bf566cb4dc7a72)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/14dcef2a7cf6da9e22425f2e241fe2e6 for a full test report.

**Comment by Flamefire** [comment]
@smoors 4.0.x ECs are broken (not by this PR, they need UCX as deps)

**Comment by Flamefire** [comment]
@smoors Fixed by https://github.com/easybuilders/easybuild-easyconfigs/pull/13329

**Comment by smoors** [comment]
Test report by @smoors
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _OpenMPI-4.0.2-GCC-9.2.0-2.32.eb_ (partial log available at https://gist.github.com/1ac16d5e14b193f72fbf290fbb101666)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
node304.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/c1e26f7bb84541cceadf92df2efb1070 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _OpenMPI-4.0.0-GCC-8.2.0-2.31.1.eb_ (partial log available at https://gist.github.com/2f6c934f2f9912c3d28aad1233f944b6)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.5
See https://gist.github.com/3e6b7016272e364da2564024dd3992de for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _OpenMPI-4.0.0-GCC-8.2.0-2.31.1.eb_ (partial log available at https://gist.github.com/f0d0257e60e37c28a9c3feb746d11057)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.5
See https://gist.github.com/c62fa4b832b3649aad0054c9e312d711 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _OpenMPI-4.0.0-GCC-8.2.0-2.31.1.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.5
See https://gist.github.com/2439f90c5aa49fb2ec7683c9f20f641c for a full test report.

File: easybuild/easyblocks/o/openmpi.py [hunk 1]
```diff
@@ -164,4 +175,33 @@ def sanity_check_step(self):
...
+            if os.path.exists(src_path):
+                test_exe = os.path.join(self.builddir, 'mpi_test_' + os.path.splitext(src)[0])
+                self.log.info("Adding minimal MPI test program to sanity checks: %s", test_exe)
+
+                # Build test binary
+                build_cmd = "%s %s -o %s" % (compiler, src_path, test_exe)
+
+                # Skip running the test if chosen
+                if not build_option('mpi_tests'):
+                    continue
```
Comment (akesandgren) [inline easybuild/easyblocks/o/openmpi.py:192]: Shouldn't this rather be above the for-loop?

Comment (Flamefire) [reply easybuild/easyblocks/o/openmpi.py]: Good catch! It is actually only supposed to skip running the tests, not compiling them. Changed that in https://github.com/easybuilders/easybuild-easyblocks/pull/2444/commits/3bc992371f74554d382b4fba13b5cc1e5f8c049f

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @Flamefire!

**Pull request merged**